python/pyarrow/tests/test_substrait.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+import sys
+import pytest
+
+import pyarrow as pa
+from pyarrow.lib import tobytes
+from pyarrow.lib import ArrowInvalid
+
+try:
+    import pyarrow.substrait as substrait
+except ImportError:
+    substrait = None
+
+# Marks all of the tests in this module
+# Ignore these with pytest ... -m 'not substrait'
+pytestmark = [pytest.mark.dataset, pytest.mark.substrait]
+
+
+@pytest.mark.skipif(sys.platform == 'win32',
+                    reason="ARROW-16392: file based URI is" +
+                    " not fully supported for Windows")
+def test_run_serialized_query(tmpdir):
+    substrait_query = """
+    {
+        "relations": [
+        {"rel": {
+            "read": {
+            "base_schema": {
+                "struct": {
+                "types": [
+                            {"i64": {}}
+                        ]
+                },
+                "names": [
+                        "foo"
+                        ]
+            },
+            "local_files": {
+                "items": [
+                {
+                    "uri_file": "file://FILENAME_PLACEHOLDER"
+                }
+                ]
+            }
+            }
+        }}
+        ]
+    }
+    """
+
+    path = os.path.join(str(tmpdir), 'substrait_data.arrow')
```

Comment:
Adding it as 

```python
path = os.path.join(str(tmpdir), 'substrait_data.ipc')
```

Error (Since we don't support it). Should we resolve this here as well? Or follow this up in another PR?

```bash
aise ArrowNotImplementedError(message)
E   pyarrow.lib.ArrowNotImplementedError: substrait::ReadRel::LocalFiles::FileOrFiles::format other than FILE_FORMAT_PARQUET
E   /Users/vibhatha/github/fork/arrow/cpp/src/arrow/engine/substrait/serde.cc:73  FromProto(plan_rel.rel(), ext_set)
E   /Users/vibhatha/github/fork/arrow/cpp/src/arrow/engine/substrait/util.cc:95  engine::DeserializePlan(substrait_buffer, consumer_factory)
E   /Users/vibhatha/github/fork/arrow/cpp/src/arrow/engine/substrait/util.cc:116  executor.Init(substrait_buffer)
```

Reply:
Oops, my bad and sorry for the mis-suggestion. It looks like we definitely need Parquet for the tests right now.

Reply:
The file just has to end in `.arrow` and then it will get picked up by our short-term IPC hack.  Proper IPC support is not far off.

Reply:
Okay then it is fine, I used the `.arrow` for the moment. We can update the test case with proper support later on.
I will add a TODO.